roofit/histfactory/src/ConfigParser.cxx

```diff
@@ -260,280 +260,210 @@ std::vector< RooStats::HistFactory::Measurement > ConfigParser::GetMeasurementsF
+      if (curAttrName == "") {
+         cxcoutEHF << "Found XML attribute in Measurement with no name.\n";
+         // ADD Output Here
+         throw hf_exc();
+      } else if (curAttrName == "Name") {
+         measurement.SetName(curAttrValue.c_str());
+      } else if (curAttrName == "Lumi") {
+         measurement.SetLumi(std::stof(curAttrValue));
+      } else if (curAttrName == "LumiRelErr") {
+         measurement.SetLumiRelErr(std::stof(curAttrValue));
+      } else if (curAttrName == "BinLow") {
+         measurement.SetBinLow(std::stoi(curAttrValue));
+      } else if (curAttrName == "BinHigh") {
+         measurement.SetBinHigh(std::stoi(curAttrValue));
+      } else if (curAttrName == "Mode") {
+         cout << "\n INFO: Mode attribute is deprecated and no longer supported, will ignore\n";
+      } else if (curAttrName == "ExportOnly") {
+         measurement.SetExportOnly(CheckTrueFalse(curAttrValue, "Measurement"));
+      } else {
+         cxcoutEHF << "Found unknown XML attribute in Measurement: " << curAttrName << "\n";
+         throw hf_exc();
       }
-      //poi// measurement.SetPOI( child->GetText() );
-     AddSubStrings( measurement.GetPOIList(), child->GetText() );
-    }
-
-    else if( child->GetNodeName() == TString( "ParamSetting" ) ) {
-      TListIter paramIt = child->GetAttributes();
-      TXMLAttr* curParam = 0;
-      while( ( curParam = dynamic_cast< TXMLAttr* >( paramIt() ) ) != 0 ) {
-
-	if( curParam->GetName() == TString( "" ) ) {
-	  cxcoutEHF << "Error: Found tag attribute with no name in ParamSetting" << std::endl;
-	  throw hf_exc();
-	}
-	else if( curParam->GetName() == TString( "Const" ) ) {
-	  if(curParam->GetValue()==TString("True")){
-	    // Fix here...?
-	    if( child->GetText() == NULL ) {
-	      cxcoutEHF << "Error: node: " << child->GetName()
-			<< " has no text." << std::endl;
-	      throw hf_exc();
-	    }
-	    AddSubStrings( measurement.GetConstantParams(), child->GetText() );
-	  }
-	}
-	else if( curParam->GetName() == TString( "Val" ) ) {
-	  double val = atof(curParam->GetValue());
-	  if( child->GetText() == NULL ) {
-	    cxcoutEHF << "Error: node: " << child->GetName()
-		      << " has no text." << std::endl;
-	    throw hf_exc();
-	  }
-	  std::vector<std::string> child_nodes = GetChildrenFromString(child->GetText());
-	  for(unsigned int i = 0; i < child_nodes.size(); ++i) {
-	    measurement.SetParamValue( child_nodes.at(i), val);
-	  }
-	  // AddStringValPairToMap( measurement.GetParamValues(), val, child->GetText() );
-	}
-	else {
-	  cxcoutEHF << "Found tag attribute with unknown name in ParamSetting: "
-		    << curParam->GetName() << std::endl;
-	  throw hf_exc();
-	}
+   } // End Loop over attributes
+
+   // Then, get the properties of the children nodes
+   TXMLNode *child = node->GetChildren();
+   while (child != nullptr) {
+      const std::string childName(child->GetName()), childNodeName(child->GetNodeName()), childText(child->GetText());
```

Comment:
```suggestion
      const std::string childName(child->GetName()), childNodeName(child->GetNodeName() ? child->GetNodeName() : ""), childText(child->GetText() ? child->GetText() : "");
```

Reply:
This fixed the crash for me.

It's admittedly not very easy to find this, since it occurs when `hist2workspace` is invoked, and that happens as a subprocess of the tests that you saw failing.

Reply:
It's in the logs of the tests, but they don't have line numbers.

One other thing:
> Unfortunately, not. I rebased and force-pushed.

It might be in your `git reflog` if you are lucky.

Reply:
indeed, constructing a `std::string` from `nullptr` is undefined behaviour... oops. Fixed and pushed. Could you trigger a rebuild?

Reply:
Sure, thanks for the update!